include/arch/arm/arch/machine/gic_v3.h

```diff
@@ -353,27 +355,30 @@ static inline void set_gic_vcpu_ctrl_vmcr(uint32_t vmcr)
  */
 static inline uint32_t get_gic_vcpu_ctrl_apr(void)
 {
-    uint32_t reg;
+    uint64_t reg;
     MRS(ICH_AP1R0_EL2, reg);
-    return reg;
+    /* 64 bit register read, top 32 bits reserved */
+    return (uint32_t) reg;
```

Comment:
> Bit 63 is used for NMI when FEAT_GICv3_NMI is implemented. But I don't think ignoring it will be a problem any time soon, not sure if a guest can ever see the NMI bit set to 1 anyway.

Agreed. The change here doesn't change the current behaviour which already ignores bit 63, but it's good to confirm that this is Ok for at least now.

> Code wise there would be no downside to making apr word_t instead, it would remove `gicVCpuIface_padding`.

We could look at that, yes. I'd prefer to make that a separate change to keep the behaviour strictly unchanged in this commit.